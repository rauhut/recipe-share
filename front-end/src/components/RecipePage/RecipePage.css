ul {
  list-style-type: none;
}

.recipe {
  max-width: 1000px;
  margin: 20px auto;
  text-align: left;
}

.recipe-title {
  padding: 15px 25px;
}

.recipe-cooktime {
  padding-left: 25px;
}

.recipe-description {
  text-align: center;
}

.recipe-ingredients {
  padding: 10px 50px;
}

.ingredient-item,
.directions-item {
  padding-bottom: 20px;
}

label input {
  margin-right: 30px;
}

.recipe-image > img {
  max-width: 100%;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 15px;
}

.recipe-directions {
  padding: 0 50px;
}
