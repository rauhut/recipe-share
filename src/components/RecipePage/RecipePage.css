ul {
  list-style-type: none;
}

.top {
  display: flex;
  justify-content: space-between;
}

.recipe {
  max-width: 1000px;
  margin: 20px auto;
  text-align: left;
}

.recipe-title {
  padding: 15px 25px;
}

.recipe-cooktime {
  padding-left: 25px;
  padding-bottom: 15px;
}

.recipe-description {
  text-align: center;
}

.recipe-ingredients {
  padding: 10px 50px;
}

.ingredient-item,
.directions-item {
  padding-bottom: 20px;
}

label input {
  margin-right: 30px;
}

ol > li,
.direction-container {
  margin-left: 30px;
  position: relative;
}

ol > li:before {
  counter-increment: list;
  content: counter(list) ".";
  position: absolute;
  right: 100%;
  margin-right: 5px;
  top: 0;
}

ol {
  list-style: none;
  counter-reset: list;
}

.recipe-image > img {
  max-width: 75%;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 15px;
}

.recipe-directions {
  padding: 0 50px;
}
