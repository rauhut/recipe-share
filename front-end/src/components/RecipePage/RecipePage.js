import React, { Component } from "react";
import "./RecipePage.css";

class RecipePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipe: {
        name: "",
        cookTime: "",
        description: "",
        ingredients: [],
        steps: [],
      },
    };
  }

  componentDidMount() {
    fetch(`http://localhost:3000/recipe/${this.props.match.params.id}`, {
      method: "get",
      headers: {
        "Content-Type": "application/json",
      },
    })
      .then((response) => {
        if (response.ok) {
          return response.json();
        } else {
          throw new Error("Something went wrong when trying to get recipe");
        }
      })
      .then((recipe) => {
        this.setState({ recipe: recipe.data });
      })
      .catch((error) => console.log(error));
  }

  render() {
    const {
      name,
      cookTime,
      description,
      ingredients,
      steps,
      picture,
    } = this.state.recipe;
    return (
      <div className="recipe">
        <button
          className="button-secondary"
          onClick={() => this.props.history.push("/")}
        >
          &laquo; Back
        </button>
        <div className="recipe-title">
          <h1>{name}</h1>
        </div>
        <div className="recipe-cooktime">Total Cook Time: {cookTime}</div>
        <div className="recipe-description">{description}</div>
        <hr></hr>
        <div className="recipe-image">
          <img src={picture} alt={name} />
        </div>
        <div className="recipe-ingredients">
          <h3>Ingredients:</h3>
          <ul>
            {ingredients.map((ingredent) => {
              return (
                <li className="ingredient-item">
                  <label className="container">
                    <input type="checkbox" />
                    {ingredent}
                  </label>
                </li>
              );
            })}
          </ul>
        </div>
        <hr></hr>
        <div className="recipe-directions">
          <h3>Directions:</h3>
          <ol>
            {steps.map((step) => {
              return <li className="directions-item">{step}</li>;
            })}
          </ol>
        </div>
      </div>
    );
  }
}

export default RecipePage;
