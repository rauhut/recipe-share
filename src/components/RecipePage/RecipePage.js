import React, { Component } from "react";
import "./RecipePage.css";
import EditRecipeModal from "../AddRecipeModal/EditRecipeModal";
import {
  Spinner,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter,
} from "reactstrap";
import defaultImage from "../../Default.jpg";

class RecipePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipe: {
        name: "",
        cookTime: "",
        description: "",
        ingredients: [""],
        steps: [""],
        picture: "",
        _id: "",
      },
      isLoading: true,
      deleteModal: false,
      editModal: false,
    };
  }

  componentDidMount() {
    fetch(
      `https://recipe-share-backend.herokuapp.com/recipe/${this.props.match.params.id}`,
      {
        method: "get",
        headers: {
          "Content-Type": "application/json",
        },
      }
    )
      .then((response) => {
        if (response.ok) {
          return response.json();
        } else {
          throw new Error("Something went wrong when trying to get recipe");
        }
      })
      .then((recipe) => {
        this.setState({ recipe: recipe.data });
        this.setState({ isLoading: false });
      })
      .catch((error) => console.log(error));
  }

  toggleDelete = () => {
    this.setState({ deleteModal: !this.state.deleteModal });
  };

  toggleEdit = () => {
    this.setState({ editModal: !this.state.editModal });
  };

  navToRecipe = (id) => {
    this.props.history.push(`/recipe/${id}`);
  };

  deleteRecipe(id) {
    const token = window.sessionStorage.getItem("token");
    fetch(`https://recipe-share-backend.herokuapp.com/recipe/${id}`, {
      method: "delete",
      headers: {
        "Content-Type": "application/json",
        Authorization: token,
      },
    })
      .then((response) => {
        if (response.ok) {
          this.props.history.push("/");
        } else {
          throw new Error("Something went wront with trying to delete recipe");
        }
      })
      .catch((error) => console.log(error));
  }

  render() {
    const {
      name,
      cookTime,
      description,
      ingredients,
      steps,
      picture,
      createdBy,
      _id,
    } = this.state.recipe;
    const { deleteModal, editModal, isLoading, recipe } = this.state;
    return (
      <div>
        {isLoading ? (
          <Spinner className="spinner" color="danger" />
        ) : (
          <div className="recipe">
            <div className="top">
              <button
                className="button-secondary"
                onClick={() => this.props.history.push("/")}
              >
                &laquo; Back
              </button>
              {createdBy === this.props.user.id && (
                <div className="modify-btns">
                  <div className="delete-btn">
                    <button
                      className="button-secondary"
                      id="delete-btn"
                      onClick={this.toggleDelete}
                    >
                      Delete
                    </button>
                    <Modal isOpen={deleteModal} toggle={this.toggleDelete}>
                      <ModalHeader toggle={this.toggleDelete}>
                        Delete "{name}"?
                      </ModalHeader>
                      <ModalBody>
                        Are you sure that you want to delete this recipe?
                      </ModalBody>
                      <ModalFooter>
                        <button
                          className="button-secondary"
                          onClick={() => this.deleteRecipe(_id)}
                        >
                          Delete
                        </button>{" "}
                        <button
                          className="button-primary"
                          id="cancel-btn"
                          onClick={this.toggleDelete}
                        >
                          Cancel
                        </button>
                      </ModalFooter>
                    </Modal>
                  </div>
                  <div className="edit-btn">
                    <button
                      className="button-secondary"
                      id="edit-btn"
                      onClick={this.toggleEdit}
                    >
                      Edit
                    </button>
                  </div>
                  <Modal isOpen={editModal} toggle={this.toggleEdit}>
                    <EditRecipeModal
                      toggle={this.toggleEdit}
                      navToRecipe={this.navToRecipe}
                      recipe={recipe}
                    ></EditRecipeModal>
                  </Modal>
                </div>
              )}
            </div>
            <div className="recipe-title">
              <h1>{name}</h1>
            </div>
            <div className="recipe-cooktime">Total Cook Time: {cookTime}</div>
            <div className="recipe-description">{description}</div>
            <hr></hr>
            <div className="recipe-image">
              <img
                src={picture}
                alt={name}
                onError={(e) => {
                  e.target.onerror = null;
                  e.target.src = defaultImage;
                }}
              />
            </div>
            <div className="recipe-ingredients">
              <h3>Ingredients</h3>
              <ul>
                {ingredients.map((ingredent, i) => {
                  return (
                    <li key={i} className="ingredient-item">
                      <label className="container">
                        <input type="checkbox" />
                        {ingredent}
                      </label>
                    </li>
                  );
                })}
              </ul>
            </div>
            <hr></hr>
            <div className="recipe-directions">
              <h3>Directions</h3>
              <ol>
                {steps.map((step, i) => {
                  return (
                    <li key={i} className="directions-item">
                      <label className="direction-container">{step} </label>
                    </li>
                  );
                })}
              </ol>
            </div>
          </div>
        )}
      </div>
    );
  }
}

export default RecipePage;
