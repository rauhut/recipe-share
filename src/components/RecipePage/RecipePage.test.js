import React from "react";
import { shallow } from "enzyme";
import RecipePage from "./RecipePage";

describe("RecipePage", () => {
  let wrapper;

  const mockRecipe = {
    name: "meatballs",
    ingredients: ["ground beef", "eggs", "bread crumbs", "garlic", "onion"],
    steps: ["step 1", "step 2", "step 3"],
    cookTime: "30 mins",
    description: "just basic meatballs",
    createdBy: "1234567890",
    _id: "5678",
  };
  const mockUser = {
    id: "1234567890",
    username: "testUser",
  };
  const mockMatch = {
    params: {
      id: "5678",
    },
  };
  const mockProps = {
    user: mockUser,
    match: mockMatch,
  };

  it("renders loading state without crashing", () => {
    expect(shallow(<RecipePage {...mockProps} />).debug()).toMatchSnapshot();
  });

  it("renders component after load without crashing", () => {
    wrapper = shallow(<RecipePage {...mockProps} />);
    wrapper.setState({ isLoading: false, recipe: mockRecipe });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("renders component with delete recipe modal open", () => {
    wrapper = shallow(<RecipePage {...mockProps} />);
    wrapper.setState({ modal: true, isLoading: false, recipe: mockRecipe });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("opens delete modal when delete button is clicked", () => {
    wrapper = shallow(<RecipePage {...mockProps} />);
    wrapper.setState({ isLoading: false, recipe: mockRecipe });
    wrapper.find('[id="delete-btn"]').simulate("click");
    expect(wrapper.state("modal")).toEqual(true);
  });

  it("closes delete modal when cancel button is clicked", () => {
    wrapper = shallow(<RecipePage {...mockProps} />);
    wrapper.setState({ isLoading: false, recipe: mockRecipe, modal: true });
    wrapper.find('[id="cancel-btn"]').simulate("click");
    expect(wrapper.state("modal")).toEqual(false);
  });
});
