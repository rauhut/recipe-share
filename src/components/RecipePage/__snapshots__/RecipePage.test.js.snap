// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage renders component after load without crashing 1`] = `
"<div>
  <div className=\\"recipe\\">
    <div className=\\"top\\">
      <button className=\\"button-secondary\\" onClick={[Function: onClick]}>
        « Back
      </button>
      <div className=\\"delete-btn\\">
        <button className=\\"button-secondary\\" id=\\"delete-btn\\" onClick={[Function (anonymous)]}>
          Delete
        </button>
        <Modal isOpen={false} toggle={[Function (anonymous)]} autoFocus={true} centered={false} scrollable={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} unmountOnClose={true} returnFocusAfterClose={true} container=\\"body\\">
          <ModalHeader toggle={[Function (anonymous)]} tag=\\"h5\\" wrapTag=\\"div\\" closeAriaLabel=\\"Close\\" charCode={215}>
            Delete &quot;
            meatballs
            &quot;?
          </ModalHeader>
          <ModalBody tag=\\"div\\">
            Are you sure that you want to delete this recipe?
          </ModalBody>
          <ModalFooter tag=\\"div\\">
            <button className=\\"button-secondary\\" onClick={[Function: onClick]}>
              Delete
            </button>
             
            <button className=\\"button-primary\\" id=\\"cancel-btn\\" onClick={[Function (anonymous)]}>
              Cancel
            </button>
          </ModalFooter>
        </Modal>
      </div>
    </div>
    <div className=\\"recipe-title\\">
      <h1>
        meatballs
      </h1>
    </div>
    <div className=\\"recipe-cooktime\\">
      Total Cook Time: 
      30 mins
    </div>
    <div className=\\"recipe-description\\">
      just basic meatballs
    </div>
    <hr />
    <div className=\\"recipe-image\\">
      <img src={[undefined]} alt=\\"meatballs\\" />
    </div>
    <div className=\\"recipe-ingredients\\">
      <h3>
        Ingredients
      </h3>
      <ul>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            ground beef
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            eggs
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            bread crumbs
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            garlic
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            onion
          </label>
        </li>
      </ul>
    </div>
    <hr />
    <div className=\\"recipe-directions\\">
      <h3>
        Directions
      </h3>
      <ol>
        <li className=\\"directions-item\\">
          <label className=\\"direction-container\\">
            step 1
             
          </label>
        </li>
        <li className=\\"directions-item\\">
          <label className=\\"direction-container\\">
            step 2
             
          </label>
        </li>
        <li className=\\"directions-item\\">
          <label className=\\"direction-container\\">
            step 3
             
          </label>
        </li>
      </ol>
    </div>
  </div>
</div>"
`;

exports[`RecipePage renders component with delete recipe modal open 1`] = `
"<div>
  <div className=\\"recipe\\">
    <div className=\\"top\\">
      <button className=\\"button-secondary\\" onClick={[Function: onClick]}>
        « Back
      </button>
      <div className=\\"delete-btn\\">
        <button className=\\"button-secondary\\" id=\\"delete-btn\\" onClick={[Function (anonymous)]}>
          Delete
        </button>
        <Modal isOpen={true} toggle={[Function (anonymous)]} autoFocus={true} centered={false} scrollable={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} unmountOnClose={true} returnFocusAfterClose={true} container=\\"body\\">
          <ModalHeader toggle={[Function (anonymous)]} tag=\\"h5\\" wrapTag=\\"div\\" closeAriaLabel=\\"Close\\" charCode={215}>
            Delete &quot;
            meatballs
            &quot;?
          </ModalHeader>
          <ModalBody tag=\\"div\\">
            Are you sure that you want to delete this recipe?
          </ModalBody>
          <ModalFooter tag=\\"div\\">
            <button className=\\"button-secondary\\" onClick={[Function: onClick]}>
              Delete
            </button>
             
            <button className=\\"button-primary\\" id=\\"cancel-btn\\" onClick={[Function (anonymous)]}>
              Cancel
            </button>
          </ModalFooter>
        </Modal>
      </div>
    </div>
    <div className=\\"recipe-title\\">
      <h1>
        meatballs
      </h1>
    </div>
    <div className=\\"recipe-cooktime\\">
      Total Cook Time: 
      30 mins
    </div>
    <div className=\\"recipe-description\\">
      just basic meatballs
    </div>
    <hr />
    <div className=\\"recipe-image\\">
      <img src={[undefined]} alt=\\"meatballs\\" />
    </div>
    <div className=\\"recipe-ingredients\\">
      <h3>
        Ingredients
      </h3>
      <ul>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            ground beef
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            eggs
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            bread crumbs
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            garlic
          </label>
        </li>
        <li className=\\"ingredient-item\\">
          <label className=\\"container\\">
            <input type=\\"checkbox\\" />
            onion
          </label>
        </li>
      </ul>
    </div>
    <hr />
    <div className=\\"recipe-directions\\">
      <h3>
        Directions
      </h3>
      <ol>
        <li className=\\"directions-item\\">
          <label className=\\"direction-container\\">
            step 1
             
          </label>
        </li>
        <li className=\\"directions-item\\">
          <label className=\\"direction-container\\">
            step 2
             
          </label>
        </li>
        <li className=\\"directions-item\\">
          <label className=\\"direction-container\\">
            step 3
             
          </label>
        </li>
      </ol>
    </div>
  </div>
</div>"
`;

exports[`RecipePage renders loading state without crashing 1`] = `
"<div>
  <Spinner className=\\"spinner\\" color=\\"danger\\" tag=\\"div\\" type=\\"border\\">
    Loading...
  </Spinner>
</div>"
`;
